tests/datasets/test_dataset.py

```diff
@@ -40,17 +46,224 @@ def test_invalid_uris(uri):
+    records = session.scalars(select(DatasetDagRunQueue)).all()
+    dag_statuses = defaultdict(lambda: defaultdict(bool))
+    for record in records:
+        dag_statuses[record.target_dag_id][record.dataset.uri] = True
+
+    serialized_dags = session.execute(
+        select(SerializedDagModel).where(SerializedDagModel.dag_id.in_(dag_statuses.keys()))
+    ).fetchall()
+
+    for (serialized_dag,) in serialized_dags:
+        dag = SerializedDAG.deserialize(serialized_dag.data)
+        for dataset_uri, status in dag_statuses[dag.dag_id].items():
+            assert dag.dataset_triggers.evaluate({dataset_uri: status}), "DAG trigger evaluation failed"
+
+
+@pytest.mark.db_test
+@pytest.mark.usefixtures("create_test_datasets")
+def test_additional_dag_with_no_triggers(dag_maker):
+    # Create an additional DAG to ensure it's not affected by dataset triggers
+    with dag_maker(dag_id="dag2"):
+        EmptyOperator(task_id="hello2")
+
+
+@pytest.fixture
+def setup_datasets_and_models(session):
+    """Fixture to create datasets and corresponding models."""
+    # Create Dataset instances
+    d1 = Dataset(uri="hello1")
+    d2 = Dataset(uri="hello2")
+
+    # Create and add DatasetModel instances to the session
+    dm1 = DatasetModel(uri=d1.uri)
+    dm2 = DatasetModel(uri=d2.uri)
+    session.add_all([dm1, dm2])
+    session.commit()
+
+    return d1, d2
+
+
+@pytest.mark.db_test
+def test_dag_with_complex_dataset_triggers(session, dag_maker, setup_datasets_and_models):
+    d1, d2 = setup_datasets_and_models
+
+    # Setup a DAG with complex dataset triggers (DatasetAny with DatasetAll)
+    with dag_maker(schedule=DatasetAny(d1, DatasetAll(d2, d1))) as dag:
+        EmptyOperator(task_id="hello")
+
+    assert isinstance(
+        dag.dataset_triggers, DatasetAny
+    ), "DAG's dataset trigger should be an instance of DatasetAny"
+    assert any(
+        isinstance(trigger, DatasetAll) for trigger in dag.dataset_triggers.objects
+    ), "DAG's dataset trigger should include DatasetAll"
+
+    serialized_triggers = SerializedDAG.serialize(dag.dataset_triggers)
+
+    deserialized_triggers = SerializedDAG.deserialize(serialized_triggers)
+
+    assert isinstance(
+        deserialized_triggers, DatasetAny
+    ), "Deserialized triggers should be an instance of DatasetAny"
+    assert any(
+        isinstance(trigger, DatasetAll) for trigger in deserialized_triggers.objects
+    ), "Deserialized triggers should include DatasetAll"
+
+    serialized_dag_dict = SerializedDAG.to_dict(dag)["dag"]
+    assert "dataset_triggers" in serialized_dag_dict, "Serialized DAG should contain 'dataset_triggers'"
+    assert isinstance(
+        serialized_dag_dict["dataset_triggers"], dict
+    ), "Serialized 'dataset_triggers' should be a dict"
+
+
+@pytest.fixture(autouse=True)
+def clear_datasets():
+    from tests.test_utils.db import clear_db_datasets
+
+    clear_db_datasets()
```

Comment:
```suggestion
    clear_db_datasets()
    yield
    clear_db_datasets()
```

Should we also clear when we finish up?